lib/src/main/java/com/cloudbees/groovy/cps/green/GreenThread.java

```diff
@@ -0,0 +1,199 @@
+package com.cloudbees.groovy.cps.green;
+
+import com.cloudbees.groovy.cps.Block;
+import com.cloudbees.groovy.cps.Continuable;
+import com.cloudbees.groovy.cps.impl.ConstantBlock;
+import com.cloudbees.groovy.cps.impl.CpsCallableInvocation;
+import com.cloudbees.groovy.cps.Outcome;
+import com.cloudbees.groovy.cps.impl.ThrowBlock;
+import java.io.Serializable;
+
+/**
+ * Represents a green thread.
+ *
+ * @author Kohsuke Kawaguchi
+ */
+public abstract class GreenThread implements Runnable, Serializable {
```

Comment:
SpotBugs didn't like that `GreenThreadState` is serializable and this class was not since `GreenThreadState.g` is a `GreenThread`. I added `Serializable`, but I don't think the `green` package is used at all, maybe it should just be deleted?

Reply:
I do not think this package was ever used. Looks like it can be deleted.